SpatialGDK/Source/SpatialGDK/Private/SpatialGDKSettings.cpp

```diff
@@ -94,8 +94,38 @@ void CheckCmdLineOverrideOptionalStringWithCallback(const TCHAR* CommandLine, co
 	UE_LOG(LogSpatialGDKSettings, Log, TEXT("%s is %s."), PrettyName,
 		   OverrideValue.IsSet() ? *(OverrideValue.GetValue()) : TEXT("not set"));
 }
+
 } // namespace
 
+#if WITH_EDITOR
+void UEventTracingSamplingSettings::PostEditChangeProperty(struct FPropertyChangedEvent& PropertyChangedEvent)
+{
+	auto IsQueryValid = [](const char* QueryStr) -> bool {
+		return SpatialGDK::TraceQueryPtr(Trace_ParseSimpleQuery(QueryStr)).Get() != nullptr;
+	};
+	Super::PostEditChangeProperty(PropertyChangedEvent);
+	const FName Name = (PropertyChangedEvent.MemberProperty != nullptr) ? PropertyChangedEvent.MemberProperty->GetFName() : NAME_None;
+	if (Name == GET_MEMBER_NAME_CHECKED(UEventTracingSamplingSettings, EventPreFilter))
+	{
+		if (!IsQueryValid(TCHAR_TO_ANSI(*EventPreFilter)))
+		{
+			FMessageDialog::Open(EAppMsgType::Ok,
```

Comment:
Nice stuff!

Reply:
I reworked btw to move the dialog in the lambda - lost the original comment!